test/mount/extensions/common_deps_test.clj

```diff
@@ -0,0 +1,69 @@
+(ns mount.extensions.common-deps-test
+  (:require
+   [clojure.test :refer [deftest testing is]]
+   [mount.extensions.common-deps :as sut]))
+
+(def system "system")
+
+(def graphs
+  {:dependencies {::overrides   #{::overrides-a ::overrides-b}
+                  ::config-a    #{}
+                  ::overrides-a #{::config-a}
+                  ::config-b    #{}
+                  ::registry    #{}
+                  ::config      #{::overrides ::env}
+                  ::system      #{::config ::debug ::db}
+                  ::debug       #{}
+                  ::db          #{::config}
+                  ::env         #{}
+                  ::overrides-b #{::config-b}}
+   :dependents   {::overrides   #{::config}
+                  ::config-a    #{::overrides-a}
+                  ::overrides-a #{::overrides}
+                  ::config-b    #{::overrides-b}
+                  ::registry    #{}
+                  ::config      #{::system ::db}
+                  ::system      #{}
+                  ::debug       #{::system}
+                  ::db          #{::system}
+                  ::env         #{::config}
+                  ::overrides-b #{::overrides}}})
```

Comment:
This is a real-world dependency graph from an application that I'm working on, with the names changed to protect the innocent 😄

Reply:
Not sure if this is as clear as it could be, with these names.